model/textparse/protobufparse.go

```diff
@@ -411,6 +411,14 @@ func (p *ProtobufParser) Next() (Entry, error) {
 			p.metricPos++
 			p.fieldPos = -2
 			p.fieldsDone = false
+			// If this is a metric family containing native
+			// histograms, we have to switch back to native
+			// histograms after parsing a classic histogram.
+			if p.state == EntrySeries &&
+				(t == dto.MetricType_HISTOGRAM || t == dto.MetricType_GAUGE_HISTOGRAM) &&
+				isNativeHistogram(p.mf.GetMetric()[0].GetHistogram()) {
```

Comment:
Nit: I see we are already using the first item to decide the type, which I think is assuming something that protobuf does not guarantee, but it's used in the code above as well so not a new problem. Maybe move to its own inline function?

Reply:
In line 356, we skip empty metric families. So we can never reach the states `EntryHistogram` or `EntrySeries` if there is not at least one metric.